sdk/core/azure-core-amqp/src/amqp/private/message_sender_impl.hpp

```diff
@@ -5,34 +5,41 @@
 
 #include "azure/core/amqp/message_sender.hpp"
 #include "claims_based_security_impl.hpp"
+#include "link_impl.hpp"
 #include <azure_uamqp_c/message_sender.h>
 #include <tuple>
 
+template <> struct Azure::Core::_internal::UniqueHandleHelper<MESSAGE_SENDER_INSTANCE_TAG>
+{
+  static void FreeMessageSender(MESSAGE_SENDER_HANDLE obj);
+
+  using type
+      = Azure::Core::_internal::BasicUniqueHandle<MESSAGE_SENDER_INSTANCE_TAG, FreeMessageSender>;
+};
+
 namespace Azure { namespace Core { namespace Amqp { namespace _detail {
+  using UniqueMessageSender = Azure::Core::_internal::UniqueHandle<MESSAGE_SENDER_INSTANCE_TAG>;
+
+  class MessageSenderFactory {
```

Comment:
I also seen some non-final classes sitting inside cpp files, i.e. with very limited visibility, this instance included. I am tempted to suggest final and constructor/destructor deletion, but I am afraid it would feel like I am really nit-picking here :)

You have applied a lot of non-critical feedback in this PR already - thank you for that! With minmalistic namespace qualifications, it is probably just me, but the code is really easier to read, looks cleaner; maybe counter-intuitively, but easier to see where the stuff comes from.

Reply:
FIxed.

Reply:
All the classes without final now have final.

Reply:
Oh, and as far as I'm concerned EVERY piece of feedback is critical - there is no such thing as "non-critical" feedback. Some of it may be minor, some of it may be major. But if the code reviewer spent the time to write a comment, it means that they want the change made. If they didn't want the change to be made, they wouldn't have left a comment.

Some of the pull request I disagree with, and I will chose to implement it or not. Others I don't feel strongly about, so it doesn't bother me to implement it.